### ISSUE TITLE
error: could not compile `parquet`

### ISSUE THREAD

**Issue by gangliao** [apache/arrow#9144]
```rust
error[E0063]: missing fields `file_offset`, `ordinal`, `total_compressed_size` in initializer of `RowGroup`
   --> /home/scq/.cargo/git/checkouts/arrow-c68ed0db3d7d4cdb/a8e2a6e/rust/parquet/src/file/metadata.rs:262:9
    |
262 |         RowGroup {
    |         ^^^^^^^^ missing `file_offset`, `ordinal`, `total_compressed_size`

error[E0063]: missing field `bloom_filter_offset` in initializer of `ColumnMetaData`
   --> /home/scq/.cargo/git/checkouts/arrow-c68ed0db3d7d4cdb/a8e2a6e/rust/parquet/src/file/metadata.rs:501:31
    |
501 |         let column_metadata = ColumnMetaData {
    |                               ^^^^^^^^^^^^^^ missing `bloom_filter_offset`

error[E0063]: missing fields `crypto_metadata`, `encrypted_column_metadata` in initializer of `ColumnChunk`
   --> /home/scq/.cargo/git/checkouts/arrow-c68ed0db3d7d4cdb/a8e2a6e/rust/parquet/src/file/metadata.rs:517:9
    |
517 |         ColumnChunk {
    |         ^^^^^^^^^^^ missing `crypto_metadata`, `encrypted_column_metadata`

error[E0063]: missing fields `encryption_algorithm`, `footer_signing_key_metadata` in initializer of `parquet_format::FileMetaData`
   --> /home/scq/.cargo/git/checkouts/arrow-c68ed0db3d7d4cdb/a8e2a6e/rust/parquet/src/file/writer.rs:178:29
    |
178 |         let file_metadata = parquet::FileMetaData {
    |                             ^^^^^^^^^^^^^^^^^^^^^ missing `encryption_algorithm`, `footer_signing_key_metadata`

   Compiling hyper v0.13.9
error: aborting due to 4 previous errors

For more information about this error, try `rustc --explain E0063`.
error: could not compile `parquet`

To learn more, run the command again with --verbose.
warning: build failed, waiting for other jobs to finish...
error: build failed
```

**Comment by gangliao** [comment]
```shell
  Compiling lazy_static v1.4.0
   Compiling cfg-if v0.1.10
   Compiling cfg-if v1.0.0
   Compiling futures-core v0.3.9
   Compiling slab v0.4.2
   Compiling fnv v1.0.7
   Compiling itoa v0.4.7
   Compiling byteorder v1.4.2
   Compiling pin-project-lite v0.1.11
   Compiling ahash v0.4.7
   Compiling pin-project-lite v0.2.1
   Compiling once_cell v1.5.2
   Compiling tower-service v0.3.0
   Compiling futures-sink v0.3.9
   Compiling pin-utils v0.1.0
   Compiling futures-io v0.3.9
   Compiling static_assertions v1.1.0
   Compiling unicode-width v0.1.8
   Compiling ppv-lite86 v0.2.10
   Compiling bytes v1.0.0
   Compiling tower-layer v0.3.1
   Compiling encode_unicode v0.3.6
   Compiling arrayvec v0.5.2
   Compiling smallvec v1.6.1
   Compiling regex-syntax v0.6.22
   Compiling stable_deref_trait v1.2.0
   Compiling hex v0.4.2
   Compiling alloc-no-stdlib v2.0.1
   Compiling nom v2.2.1
   Compiling try-lock v0.2.3
   Compiling void v1.0.2
   Compiling adler v0.2.3
   Compiling httpdate v0.3.2
   Compiling integer-encoding v1.1.5
   Compiling unicode-segmentation v1.7.1
   Compiling scopeguard v1.1.0
   Compiling strsim v0.8.0
   Compiling utf8parse v0.2.0
   Compiling ansi_term v0.11.0
   Compiling vec_map v0.8.2
   Compiling base64 v0.12.3
   Compiling base64 v0.13.0
   Compiling percent-encoding v2.1.0
   Compiling metatype v0.2.0
   Compiling more-asserts v0.2.1
   Compiling json v0.12.4
   Compiling libc v0.2.82
   Compiling serde v1.0.118
   Compiling memchr v2.3.4
   Compiling log v0.4.11
   Compiling getrandom v0.1.16
   Compiling bitflags v1.2.1
   Compiling tracing-core v0.1.17
   Compiling thread_local v1.1.0
   Compiling sharded-slab v0.1.1
   Compiling ryu v1.0.5
   Compiling proc-macro-nested v0.1.6
   Compiling serde_json v1.0.61
   Compiling typenum v1.12.0
   Compiling futures-channel v0.3.9
   Compiling futures-task v0.3.9
   Compiling twox-hash v1.6.0
   Compiling hashbrown v0.9.1
   Compiling regex-automata v0.1.9
   Compiling hash32 v0.1.1
   Compiling lexical-core v0.7.4
   Compiling textwrap v0.11.0
   Compiling heapless v0.5.6
   Compiling httparse v1.3.4
   Compiling nix v0.15.0
   Compiling alloc-stdlib v0.2.1
   Compiling build_id v0.2.1
   Compiling http v0.2.3
   Compiling crc32fast v1.2.1
   Compiling getrandom v0.2.1
   Compiling inventory v0.1.10
   Compiling snap v1.0.3
   Compiling parquet v3.0.0-SNAPSHOT (https://github.com/DSLAM-UMD/arrow?branch=scq#a8e2a6e2)
   Compiling heck v0.3.2
   Compiling nix v0.18.0
   Compiling indexmap v1.6.1
   Compiling num-traits v0.2.14
   Compiling num-integer v0.1.44
   Compiling num-bigint v0.3.1
   Compiling num-rational v0.3.2
   Compiling num-iter v0.1.42
   Compiling crossbeam-utils v0.8.1
   Compiling memoffset v0.6.1
   Compiling miniz_oxide v0.4.3
   Compiling generic-array v0.14.4
   Compiling ahash v0.6.2
   Compiling arrow v3.0.0-SNAPSHOT (https://github.com/DSLAM-UMD/arrow?branch=scq#a8e2a6e2)
   Compiling relative v0.2.2
   Compiling brotli-decompressor v2.3.1
   Compiling tracing-subscriber v0.2.15
   Compiling csv-core v0.1.10
   Compiling aho-corasick v0.7.15
   Compiling want v0.3.0
   Compiling sqlparser v0.6.1
   Compiling generic-array v0.13.2
   Compiling generic-array v0.12.3
   Compiling num_cpus v1.13.0
   Compiling net2 v0.2.37
   Compiling iovec v0.1.4
   Compiling signal-hook-registry v1.3.0
   Compiling atty v0.2.14
   Compiling time v0.1.43
   Compiling dirs v1.0.5
   Compiling dirs-sys-next v0.1.1
   Compiling socket2 v0.3.19
   Compiling fs2 v0.4.3
   Compiling nix v0.19.1
   Compiling brotli v3.3.0
   Compiling flate2 v1.0.19
   Compiling num-complex v0.3.1
   Compiling ordered-float v1.1.1
   Compiling procinfo v0.4.2
   Compiling serde_closure v0.3.2
   Compiling crossbeam-epoch v0.9.1
   Compiling crossbeam-queue v0.3.1
   Compiling crossbeam-channel v0.5.0
   Compiling threadpool v1.8.1
   Compiling regex v1.4.3
   Compiling mio v0.6.23
   Compiling as-slice v0.1.4
   Compiling rand_core v0.5.1
   Compiling clap v2.33.3
   Compiling term v0.5.2
   Compiling dirs-next v2.0.0
   Compiling zstd-sys v1.4.19+zstd.1.4.8
   Compiling lz4-sys v1.9.2
   Compiling pin-project v0.4.27
   Compiling futures-util v0.3.9
   Compiling tracing v0.1.22
   Compiling thiserror v1.0.23
   Compiling pin-project v1.0.3
   Compiling async-stream v0.2.1
   Compiling crossbeam-deque v0.8.0
   Compiling thrift v0.13.0
   Compiling rand_chacha v0.2.2
   Compiling rand_pcg v0.2.1
   Compiling mio-uds v0.6.8
   Compiling rustyline v7.1.0
   Compiling ctrlc v3.1.7
   Compiling tower-discover v0.3.0
   Compiling tower-load-shed v0.3.0
   Compiling tracing-futures v0.2.4
   Compiling tracing-error v0.1.2
   Compiling flatbuffers v0.8.0
   Compiling crossbeam v0.8.0
   Compiling parquet-format v2.7.0
   Compiling rand v0.7.3
   Compiling futures-executor v0.3.9
   Compiling tower-util v0.3.1
   Compiling palaver v0.2.8
   Compiling bytes v0.5.6
   Compiling bstr v0.2.14
   Compiling chrono v0.4.19
   Compiling erased-serde v0.3.13
   Compiling uuid v0.8.1
   Compiling http-serde v1.0.1
   Compiling serde_bytes v0.11.5
   Compiling num v0.3.1
   Compiling futures v0.3.9
   Compiling tokio v0.2.24
   Compiling http-body v0.3.1
   Compiling prost v0.6.1
   Compiling csv v1.1.5
   Compiling sqlparser v0.7.0
   Compiling serde_with v1.6.0
   Compiling typetag v0.1.6
   Compiling aws_lambda_events v0.4.0
   Compiling prettytable-rs v0.8.0
   Compiling serde_traitobject v0.2.7
   Compiling arrow-flight v3.0.0-SNAPSHOT (https://github.com/DSLAM-UMD/arrow?branch=scq#a8e2a6e2)
   Compiling lz4 v1.23.2
   Compiling zstd-safe v3.0.0+zstd.1.4.8
   Compiling zstd v0.6.0+zstd.1.4.8
   Compiling tokio-util v0.3.1
   Compiling tower-load v0.3.0
   Compiling tower-buffer v0.3.0
   Compiling tower-make v0.3.0
   Compiling tower-ready-cache v0.3.1
   Compiling tower-timeout v0.3.0
   Compiling tower-retry v0.3.0
   Compiling dataflow v0.1.0 (/home/gangliao/ci/actions-runner/_work/ServerlessCQ/ServerlessCQ/src/dataflow)
   Compiling tower-limit v0.3.1
   Compiling h2 v0.2.7
   Compiling tower-balance v0.3.0
   Compiling tower v0.3.1
   Compiling common v0.1.0 (/home/gangliao/ci/actions-runner/_work/ServerlessCQ/ServerlessCQ/src/common)
   Compiling hyper v0.13.9
```

==>

Why did cargo compile parquet-format v2.7.0?  We need parquet-format v2.6.1.

**Comment by sunchao** [comment]
@gangliao this should be fixed by #9138. Can you try again to see if the issue is gone?

**Comment by gangliao** [comment]
Thanks. @sunchao
